### PR TITLE
Shift select implementation

### PR DESCRIPTION
**Pull request by mozkomor05**
Allow user to select multiple records in `record/list` view by holding `Shift` key.

### CODE REVIEW

**Comment by yurikuzn** [comment]
Interesting idea. I will take a look.

v7.2 is frozen for feature adding, so I think we'd rather address it to v7.3.

Could also accept a CLA please. https://github.com/espocrm/cla

**Comment by yurikuzn** [comment]
Maybe would be reasonable to use document.activeElement instead of storing last clicked element. Not sure yet, need to try first.

File: client/src/views/record/list.js [hunk 1]
```diff
@@ -505,15 +510,31 @@ function (Dep, MassActionHelper, ExportHelper) {
              * @this module:views/record/list.Class
              */
             'click .record-checkbox': function (e) {
-                let $target = $(e.currentTarget);
+                const $target = $(e.currentTarget);
+
+                if (!this._lastChecked) {
```
Comment (yurikuzn) [inline client/src/views/record/list.js:515]: This if-block cause an issue when checked the first time that the item is not actually selected. Seems it's added by a mistake and should be removed. Then the following if to be changed to

```
if (e.shiftKey && this._lastChecked) {
```

Comment (mozkomor05) [reply client/src/views/record/list.js:515]: Good point, missed that.

**Pull request merged**